backends/cadence/reference/operators/quantized_ops.h

```diff
@@ -0,0 +1,233 @@
+  void BINARY_FUNC_NAME(                                                      \
+      const ::executorch::aten::Tensor& X,                                    \
+      float X_scale,                                                          \
+      int32_t X_zero_point,                                                   \
+      const ::executorch::aten::Tensor& Y,                                    \
+      float Y_scale,                                                          \
+      int32_t Y_zero_point,                                                   \
+      float out_scale,                                                        \
+      int32_t out_zero_point,                                                 \
+      ::executorch::aten::Tensor& out) {                                      \
+    const T* __restrict__ X_data = X.const_data_ptr<T>();                     \
+    const T* __restrict__ Y_data = Y.const_data_ptr<T>();                     \
+    T* __restrict__ out_data = out.mutable_data_ptr<T>();                     \
+    size_t Y_numel = Y.numel();                                               \
+    size_t X_numel = X.numel();                                               \
+    float inv_out_scale = 1.0f / out_scale;                                   \
+    /* Tensor that has the same element of X */                               \
+    if (Y_numel == X_numel) {                                                 \
+      for (size_t i = 0; i < X_numel; ++i) {                                  \
+        float x = kernels::dequantize<T>(X_data[i], X_scale, X_zero_point);   \
+        float y = kernels::dequantize<T>(Y_data[i], Y_scale, Y_zero_point);   \
+        float z = x OP y;                                                     \
+        out_data[i] = kernels::quantize<T>(z, inv_out_scale, out_zero_point); \
+      }                                                                       \
+    } /* if Y is a scalar Tensor */                                           \
+    else if (Y_numel == 1) {                                                  \
+      float y = kernels::dequantize<T>(Y_data[0], Y_scale, Y_zero_point);     \
+      for (size_t i = 0; i < X_numel; ++i) {                                  \
+        float x = kernels::dequantize<T>(X_data[i], X_scale, X_zero_point);   \
+        float z = x OP y;                                                     \
+        out_data[i] = kernels::quantize<T>(z, inv_out_scale, out_zero_point); \
+      }                                                                       \
+    } /* other broadcasting cases */                                          \
+    else {                                                                    \
+      ET_DCHECK_MSG(false, "Unsupported broadcasting");                       \
+    }                                                                         \
+  }
```

Comment:
this macro is not used. It's only for quantized_add, and will need to be removed anyway with PT2 quant